nixos/modules/services/web-servers/h2o/default.nix

```diff
@@ -258,6 +384,67 @@ in
           ${lib.getExe cfg.package} ${lib.strings.escapeShellArgs args}
         '';
     };
-  };
 
+    # This service waits for all certificates to be available before reloading
+    # H2O configuration. `tlsTargets` are added to `wantedBy` + `before` which
+    # allows the `acme-finished-$cert.target` to signify the successful updating
+    # of certs end-to-end.
+    systemd.services.h2o-config-restart =
+      let
+        tlsTargets = map (certName: "acme-${certName}.target") certNames.all;
+        tlsServices = map (certName: "acme-${certName}.service") certNames.all;
+      in
+      mkIf (certNames.all != [ ]) {
+        wantedBy = tlsServices ++ [ "multi-user.target" ];
+        before = tlsTargets;
+        after = tlsServices;
+        unitConfig = {
+          ConditionPathExists = lib.optionals (tlsServices != [ ]) (
+            map (certName: "${certs.${certName}.directory}/fullchain.pem") certNames.all
+          );
+          # Disable rate limiting for this, since it may be triggered quickly
+          # a bunch of times if a lot of certificates are renewed in quick
+          # succession. Unlike NGINX, it seems a restart is needed over a
+          # cheaper reload, which is a bit more expensive but perhaps it is
+          # fine.
+          #
+          # FIXME: like NGINX’s FIXME, there’s probably a better way to do
+          # this.
+          StartLimitIntervalSec = 0;
+        };
+        serviceConfig = {
+          Type = "oneshot";
+          TimeoutSec = 60;
+          ExecCondition = "/run/current-system/systemd/bin/systemctl -q is-active h2o.service";
```

Comment:
I don't think I've ever seen a path like this written out in a nixos module. I don't see a problem with it, I just couldn't help but comment.